src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
Quite welcome.

I understand the confusion; it took me a while to figure things out. What is returned for the actions in librcsc is, at most, an indication as to whether it looks like it's possible for the action to succeed, because librcsc isn't "doing" the action - it's transmitting to the server what the agent would like to do.

I suppose "MAYBE" might be a bit discouraging - perhaps "POSSIBLE" for "UNKNOWN" and "LIKELY" for "MAYBE"?  I tend toward caution, understatement, and trying very hard not to promise (even implicitly) what I can't deliver.

In terms of knowing whether the action succeeded, one could figure this out for *some* of the actions by examining the changes between the prior state and the state after the action has gone in. (I say *some* because it's a lot harder to tell for, say, Move.) That is part of how I was figuring out possibly-effective actions for the chart, but it was still rather variable, and would be dependent on the agent's perceptions in any event (even using fullstate + a second agent supplying information didn't always resolve questions, so non-fullstate and without a second agent would amplify the unknowns). (Even then, given noise in actions one gets into questions like "is it a success if the kick doesn't get the ball exactly where you wanted it to go, but did get it closer than it was?") So, yes, *sometimes* it might be possible to say "GOOD", but it would take quite a bit more work to do it in C++; I actually didn't finish fully evaluating some of the actions this way in the program I wrote, and the McCabe complexity rating of the subroutine doing the main evaluation (in Python) is still 62... recommended for comprehensibility is no higher than 15, and preferably no higher than 10. (That's leaving out the code for figuring out what the current state is, for instance, which has a rating of 48. Admittedly, there are probably ways to simplify the evaluation and break it into clearer sections, but the entire program is already 1710 lines...)